pkg/operator/controllers/monitoring/monitoring_controller.go

```diff
@@ -89,33 +117,23 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		return reconcile.Result{}, err
 	}
 
-	changed := false
-	// we are disabling persistence. We use omitempty on the struct to
-	// clean the fields
-	if configData.PrometheusK8s.Retention != "" {
-		configData.PrometheusK8s.Retention = ""
-		changed = true
-	}
-	if !reflect.DeepEqual(configData.PrometheusK8s.VolumeClaimTemplate, struct{ api.MissingFields }{}) {
-		configData.PrometheusK8s.VolumeClaimTemplate = struct{ api.MissingFields }{}
-		changed = true
-	}
-
-	if !reflect.DeepEqual(configData.AlertManagerMain.VolumeClaimTemplate, struct{ api.MissingFields }{}) {
-		configData.AlertManagerMain.VolumeClaimTemplate = struct{ api.MissingFields }{}
-		changed = true
-	}
-
-	if !isCreate && !changed {
-		return reconcile.Result{}, nil
-	}
+	// Nil out the fields we don't want set
+	configData.AlertManagerMain.VolumeClaimTemplate = nil
+	configData.PrometheusK8s.Retention = nil
+	configData.PrometheusK8s.VolumeClaimTemplate = nil
 
 	var b []byte
 	err = codec.NewEncoderBytes(&b, r.jsonHandle).Encode(configData)
 	if err != nil {
 		return reconcile.Result{}, err
 	}
 
+	changed := !bytes.Equal(b, configDataJSON)
```

Comment:
Not sure that comparing json strings is reliable: you can have same data, but marshaled in different order.

If we want to exit early if `!changed` then we should be checking on go value, I think. Similarly to what we had before:

```go
if configData.PrometheusK8s.Retention != "" {
    configData.PrometheusK8s.Retention = ""
    changed = true
}

if configData.PrometheusK8s.VolumeClaimTemplate != nil {
    configData.PrometheusK8s.VolumeClaimTemplate = nil
    changed = true
}

if configData.AlertManagerMain.VolumeClaimTemplate != nil {
    configData.AlertManagerMain.VolumeClaimTemplate = nil
    changed = true
}
```

Is there a better option?

Reply:
Yes it pretty much always reported as changed == true.  Now that we're using pointers for `VolumeClaimTemplate` fields and the `json.RawMessage` we can go back to this.  

Updated.